### PR TITLE
feat: add shortcut previous button and map button

### LINKED ISSUE #714: Show map button

### LINKED ISSUE #715: Handle from qs for shortcut origin

### PR DESCRIPTION
**Pull request by pyphilia**
- map button to return to builder's map 
  - close #714 
- add shortcut from button **within player only** 
  - close #715 

Screenshot 2024-05-22 at 11 56 02

### CODE REVIEW

**Review by LinaYahya — APPROVED** [review]

File: cypress/e2e/shortcut.cy.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    );
+
+    // need wait for wrongly shown button to appear
+    cy.wait(1000);
+
+    // should not show from button
+    cy.get(`#${BACK_TO_SHORTCUT_ID}`).should('not.exist');
+  });
+
+  it.only('Hacking from url with external url is safe', () => {
```
Comment (spaenleh) [inline cypress/e2e/shortcut.cy.ts:73]: remove only ?

File: package.json [hunk 2]
```diff
@@ -38,9 +38,9 @@
     "@emotion/styled": "11.11.5",
     "@graasp/chatbox": "3.1.0",
     "@graasp/query-client": "3.8.0",
-    "@graasp/sdk": "4.10.1",
+    "@graasp/sdk": "github:graasp/graasp-sdk#qs-for-host-manager",
```
Comment (spaenleh) [inline package.json:41]: An sdk version has been created, you can update.

File: src/config/selectors.ts [hunk 3]
```diff
@@ -66,3 +66,5 @@ export const APP_NAVIGATION_PLATFORM_SWITCH_BUTTON_IDS = {
 };
 
 export const TREE_NODE_GROUP_CLASS = 'tree-node-group';
+export const BACK_TO_SHORTCUT_ID = 'backToButtonShortcut';
+export const ITEM_MAP_BUTTON_ID = 'item-map-button';
```
Comment (spaenleh) [inline src/config/selectors.ts:70]: use the same standard ? 😅

File: src/langs/fr.json [hunk 4]
```diff
@@ -28,5 +28,8 @@
   "NAVIGATION_ISLAND_CHAT_BUTTON_HELPER_TEXT_READERS": "La discussion n'a pas été activée pour cet élément.",
   "NAVIGATION_ISLAND_CHAT_BUTTON_HELPER_TEXT_WRITERS": "Pour activer la discussion, allez dans les paramètres de l'élément dans le Builder et activez le paramètre « Afficher le chat dans le lecteur ».",
   "NAVIGATION_ISLAND_PINNED_BUTTON_HELPER_TEXT_READERS": "Il n'y a aucun élément épinglé pour cette page.",
-  "NAVIGATION_ISLAND_PINNED_BUTTON_HELPER_TEXT_WRITERS": "Pour ajouter des éléments à la section épinglée, définissez-les comme épinglés dans l'interface Builder."
+  "NAVIGATION_ISLAND_PINNED_BUTTON_HELPER_TEXT_WRITERS": "Pour ajouter des éléments à la section épinglée, définissez-les comme épinglés dans l'interface Builder.",
+  "MAP_BUTTON_TEXT": "Voir {{name}} sur la Carte",
+  "MAP_BUTTON_DISABLED_TEXT": "Cet élément n'a pas de localisation",
+  "FROM_SHORTCUT_BUTTON_TEXT": "Retourner sur {{name}}"
```
Comment (spaenleh) [inline src/langs/fr.json:34]: Not sure...
```suggestion
  "FROM_SHORTCUT_BUTTON_TEXT": "Retourner à {{name}}"
```

File: src/modules/item/Item.tsx [hunk 5]
```diff
@@ -294,10 +296,14 @@ const ShortcutContent = ({ item }: { item: ShortcutItemType }): JSX.Element => {
 };
 
 const FolderButtonContent = ({ item }: { item: FolderItemType }) => {
+  const [searchParams] = useSearchParams();
   const { data: thumbnail } = hooks.useItemThumbnailUrl({
     id: item.id,
     size: ThumbnailSize.Medium,
   });
+  searchParams.set('from', window.location.pathname);
+  searchParams.set('fromName', item.name);
```
Comment (spaenleh) [inline src/modules/item/Item.tsx:305]: Does this modify the params of the current page ? Should we make a copy ?

File: src/modules/item/SectionHeader.tsx [hunk 6]
```diff
@@ -21,25 +21,32 @@ const SectionHeader = ({ item }: SectionHeaderProps): JSX.Element => {
...
+              {item.name}
+            </Typography>
+            <Typography variant="caption">
+              {t(PLAYER.ITEM_TITLE_UPDATED_AT, {
+                date: formatDate(item.updatedAt, {
+                  locale: i18n.language,
+                }),
+              })}
+            </Typography>
+          </Stack>
```
Comment (spaenleh) [inline src/modules/item/SectionHeader.tsx:49]: I think you did not mean to change this code. It broke the header:
Before:
Screenshot 2024-05-24 at 11 10 49
Now: 
Screenshot 2024-05-24 at 11 10 09

File: src/modules/navigationIsland/CustomButtons.tsx [hunk 7]
```diff
@@ -11,11 +11,15 @@ const baseStyle = (theme: Theme) => ({
   borderRadius: theme.spacing(1),
   // set a fixed height 8 + 8 for padding + 24 for the icon height
   height: '40px',
+  '&:hover': {
+    cursor: 'pointer',
+  },
```
Comment (spaenleh) [inline src/modules/navigationIsland/CustomButtons.tsx:16]: Nice addition

File: src/modules/item/FromShortcutButton.tsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    return null;
+  }
+
+  if (fromUrl) {
+    return (
+      <Stack direction="column" justifyContent="center" alignItems="center">
+        <Button
+          id={BACK_TO_SHORTCUT_ID}
+          component={Link}
+          to={fromUrl}
```
Comment (spaenleh) [inline src/modules/item/FromShortcutButton.tsx:33]: Does this maintain the existing query params ? like fullscreen etc ?

File: src/modules/navigationIsland/GeolocationButton.tsx [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+  const isDisabled = !geoloc;
+
+  const tooltip = isDisabled
+    ? t(PLAYER.MAP_BUTTON_DISABLED_TEXT)
+    : t(PLAYER.MAP_BUTTON_TEXT, { name: geoloc.item.name });
+
+  const component = (
+    <span id={ITEM_MAP_BUTTON_ID}>
+      <ToolButton key="chatButton" disabled={isDisabled} aria-label={tooltip}>
```
Comment (spaenleh) [inline src/modules/navigationIsland/GeolocationButton.tsx:46]:
```suggestion
      <ToolButton key="mapButton" disabled={isDisabled} aria-label={tooltip}>
```

**Review by spaenleh — CHANGES_REQUESTED** [review]
Looks good, just a few nitpicks and a layout issue. Let me know if you have questions. 🎈

**Review by ReidyT — COMMENTED** [review]
I didn't have checked the code yet, but it is wanted to have the button in orange ? Should we use graasp's primary colour instead, so that the button integrates in better with the rest of the design?

**Review by spaenleh — APPROVED** [review]
Looks good !

**Pull request merged**